File: src/c/common/write.c
#include "./write.h"
#include "./error.h"
#include "./file.h"

void *writeTime(FILE *fp, double time)
{
    if (fp == NULL)
    {
        error_file();
        return NULL;
    }
    fprintf(fp, "%.9f\n", time);
    return NULL;
}

File: src/go/common/write.go
package parallelNP

import (
	"fmt"
	"os"
	"time"
)

func WriteTime(fp *os.File, t time.Duration) {
	// Convert the duration to seconds
	seconds := t.Seconds()

	_, err := fmt.Fprintf(fp, "%.9f\n", seconds)
	if err != nil {
		IOError("write.go", "Error writing to the file", err)
	}
}
